packages/liveblocks-core/src/types/IWebSocket.ts

```diff
@@ -0,0 +1,60 @@
+export interface IWebSocketEvent {
+  type: string;
+}
+
+export interface IWebSocketCloseEvent extends IWebSocketEvent {
+  readonly code: number;
+  readonly wasClean: boolean;
+  readonly reason: string;
+}
+
+export interface IWebSocketMessageEvent extends IWebSocketEvent {
+  readonly data: string | Buffer | ArrayBuffer | readonly Buffer[];
+}
+
+export interface IWebSocketInstance {
+  readonly CONNECTING: number; // 0
+  readonly OPEN: number; // 1
+  readonly CLOSING: number; // 2
+  readonly CLOSED: number; // 3
+
+  readonly readyState: number;
+
+  addEventListener(type: "close", listener: (this: IWebSocketInstance, ev: IWebSocketCloseEvent) => unknown): void; // prettier-ignore
+  addEventListener(type: "message", listener: (this: IWebSocketInstance, ev: IWebSocketMessageEvent) => unknown): void; // prettier-ignore
+  addEventListener(type: string, listener: (this: IWebSocketInstance, ev: IWebSocketEvent) => unknown): void; // prettier-ignore
+  //                     ^^^^^^ Includes "open" and "error"
+
+  removeEventListener(type: "close", listener: (this: IWebSocketInstance, ev: IWebSocketCloseEvent) => unknown): void; // prettier-ignore
+  removeEventListener(type: "message", listener: (this: IWebSocketInstance, ev: IWebSocketMessageEvent) => unknown): void; // prettier-ignore
+  removeEventListener(type: string, listener: (this: IWebSocketInstance, ev: IWebSocketEvent) => unknown): void; // prettier-ignore
```

Comment:
instead of string, would it be better to do `"open" | "error"`?

Reply:
Good point! I initially tried doing this, but thought it didn't work, but it must have been something else. I changed it in 96f6d21ae0040bdaa2a1f400656bd6b46e933b20, and it still works. Thanks! 🙏